addon/services/asset-loader.js

```diff
@@ -237,14 +238,27 @@ export default Ember.Service.extend({
     },
 
     css(uri) {
-      return new RSVP.Promise((resolve, reject) => {
-        const link = createLoadElement('link', resolve, reject);
-
-        link.rel = 'stylesheet';
-        link.href = uri;
-
-        document.head.appendChild(link);
-      });
+      // Since onload/onerror event support is super spotty,
+      // we "pre"-fetch the stylesheet...
+      return fetch(uri).then((response) => {
+          if (!response.ok) {
+            throw new Error(`CSS asset ${uri} responsed with status ${response.status}.`);
+          }
+
+          return new RSVP.Promise((resolve, reject) => {
+            // ...then put it in the DOM so it loads "instantly" from cache.
```

Comment:
Not guaranteed to be true. Asset could be larger than available cache, could notice that it wasn't used and discard itself...